### PR TITLE
Port Andy's tests over to Nick's implementation

### PR DESCRIPTION
**Pull request by nickjalbert**
Ported your tests to my implementation.  Probably could have done something clever to reduce code repetition, but this was quick and didn't require much thought. :)

Also updated my implementation to return reward for the particular step rather than total score.

Also also, updated github to automatically lint and test pull requests.

### CODE REVIEW

File: nick_2048.py [hunk 1]
```diff
@@ -1,11 +1,21 @@
...
 
 class Nick2048:
-    UP = "U"
-    RIGHT = "R"
-    DOWN = "D"
-    LEFT = "L"
+    RIGHT = 0
+    DOWN = 1
+    LEFT = 2
+    UP = 3
```
Comment (andyk) [inline nick_2048.py]: This is great!  Almost changed this myself :)

File: scripts/lint [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
```
Comment (andyk) [inline scripts/lint]: yes lint tests!

File: scripts/lint [hunk 3]
```diff
@@ -0,0 +1,5 @@
...
+
+set -eu
+
+flake8 . --exclude=./improved-funicular/ --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
```
Comment (andyk) [inline scripts/lint]: I had to `pip install flake8==3.7.9` -- you want to add that to requirements.txt?

Comment (nickjalbert) [reply scripts/lint]: Ah good catch!

**Pull request merged**